Assets/LeapMotion/Modules/HierarchyRecording/Scripts/LeapRecording/VectorHandRecording/VectorFrameCurves.cs

```diff
@@ -0,0 +1,59 @@
+﻿using Leap.Unity.Query;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Leap.Unity.Recording {
+
+  /// <summary> VectorHand Animation Curve data struct for a Leap frame. </summary>
+  [System.Serializable]
+  public struct VectorFrameCurves {
+
+    /// <summary> Hand curve data for the left hand. </summary>
+    public VectorHandCurves leftHandCurves;
+
+    /// <summary> Hand curve data for the right hand. </summary>
+    public VectorHandCurves rightHandCurves;
+
+    /// <summary>
+    /// Adds keyframe data from the provided frame to this animation data at the time
+    /// specified in seconds.
+    /// 
+    /// Data is only input from the first left and first right hands. If the frame lacks
+    /// a left or a right hand, that hand is considered untracked.
+    /// </summary>
+    public void AddKeyframes(float keyframeTime, Frame frame) {
+      var leftHand  = frame.Hands.Query().FirstOrDefault(h => h.IsLeft);
+      var rightHand = frame.Hands.Query().FirstOrDefault(h => !h.IsLeft);
+
+      leftHandCurves .AddKeyframes(keyframeTime, leftHand);
+      rightHandCurves.AddKeyframes(keyframeTime, rightHand);
+    }
+
+    /// <summary>
+    /// Processes all of the current keyframe data and compresses them using
+    /// AnimationCurveUtil.
+    /// </summary>
+    public void CompressCurves() {
+      leftHandCurves .Compress();
```

Comment:
Has formatting gone too far???
Really tho, I had no idea that you could put whitespace between the object and the period for methods and stuff.

Reply:
I'll admit this is pretty egregious formatting. Sometimes I like to do it when I'm constantly doing left-handed and right-handed operations one-after-the-other throughout a file, because then I can use vertical alignment as a very fast heuristic for whether the code throughout the file is structured correctly (faster bugspotting, in other words).